File: src/main/java/com/footverse/address/dto/AddressResponse.java
package com.footverse.address.dto;

/**
 * A shipping address returned to clients (dto-spec §8).
 *
 * @param id             the address id
 * @param recipientName  the recipient name
 * @param recipientPhone the recipient phone in VN format
 * @param province       the province
 * @param district       the district
 * @param ward           the ward
 * @param streetAddress  the house number / street
 * @param isDefault      whether this is the caller's default address
 */
public record AddressResponse(
        Long id,
        String recipientName,
        String recipientPhone,
        String province,
        String district,
        String ward,
        String streetAddress,
        boolean isDefault) {
}

File: app/lib/features/address/models/address_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'address_response.g.dart';

/// A customer's shipping address (dto-spec §8). The server owns the
/// exactly-one-default-per-user invariant; the client renders [isDefault] as
/// returned and never computes it (business-rules → Shipping Address).
@JsonSerializable(createToJson: false)
class AddressResponse {
  const AddressResponse({
    required this.id,
    required this.recipientName,
    required this.recipientPhone,
    required this.province,
    required this.district,
    required this.ward,
    required this.streetAddress,
    required this.isDefault,
  });

  factory AddressResponse.fromJson(Map<String, dynamic> json) =>
      _$AddressResponseFromJson(json);

  final int id;
  final String recipientName;
  final String recipientPhone;
  final String province;
  final String district;
  final String ward;
  final String streetAddress;
  final bool isDefault;
}
